packages/backend/src/api/controllers/UserController.ts

```diff
@@ -502,9 +546,28 @@ export class UserController {
 function toUserAssetEntry(
   asset: PreprocessedAssetHistoryRecord,
   tradingMode: TradingMode,
+  escapableMap: EscapableMap,
+  freezeStatus: FreezeStatus,
   collateralAssetId?: AssetId,
   assetDetailsMap?: AssetDetailsMap
 ): UserAssetEntry {
+  const escapableEntry = escapableMap[asset.positionOrVaultId.toString()]
+  let action: UserAssetEntry['action'] = 'NO-ACTION'
+
+  if (tradingMode === 'spot' || asset.assetHashOrId === collateralAssetId) {
+    if (freezeStatus !== 'frozen') {
+      action = 'WITHDRAW'
+    } else {
+      action = escapableEntry === undefined ? 'ESCAPE' : 'NO-ACTION'
```

Comment:
don't we have convention NO_ACTION rather than NO-ACTION?

Reply:
Fixed